### PR TITLE
Adds a mapping helper that can cycle multiple airlocks regardless of location

### PR DESCRIPTION
**Pull request by AlinaStarkova**
for reference, i was told adding mapping tools was ok during feature freeze, blame mothsquares if it actually isnt *sips tea*

## About The Pull Request
currently cycle helpers only work in straight lines, this limits mapping design for airlocks, i now present to you a new mapping helper where all you do is set a var for the helper, and paste it over airlocks, and those airlocks will now all sync up so only one of them can be open at a time (as of now clicking an airlock to open them wont cycle, but the current cycle helpers dont do it either so its consistent)

heres what it looks like in sdmm, set a var and paste (and yes i also pre-added a left, and right side cycles to tram bridge!)
Capture1234

heres showing that it properly sets its list of connected airlocks
Capture2344

and heres showing that after i bashed an airklock down with a fireaxe it properly removed the destroyed airlock from its list
Capture3455

and heres it in action, as you can see, when i bump into a door on the left, the right side door remains open, they are unconnected, but if i bump into ANY door on the left, any other open door regardless of its distance or location will close

https://user-images.githubusercontent.com/40489693/127975872-c8aafe2f-fc9a-4891-bd41-2a875f23cb78.mp4

## Why It's Good For The Game
mapping tools to make better maps that dont have to follow certain standards due to constraints with the code (tram bridge now cycles, you COULD also change out brig airlock, science airlock, and the airlocks on bridge on other maps to use the multi-cycle-helper so you no longer accidently leave doors open if you enter from a diagnal path!)

not player facing, no CL, but its a slight map edit, and a qol mapping helper

### CODE REVIEW

File: code/game/machinery/doors/airlock.dm
Comment (MrStonedOne) [inline code/game/machinery/doors/airlock.dm:180]:
```suggestion
	if (cyclelinkeddir)
```

File: code/game/machinery/doors/airlock.dm
Comment (MrStonedOne) [inline code/game/machinery/doors/airlock.dm:349]:
```suggestion
	if (cyclelinkedairlock)
		if (cyclelinkedairlock.cyclelinkedairlock == src)
```

**Review by MrsTonedOne — CHANGES_REQUESTED** [review]
> #### Enforced not enforced
> The following coding styles are not only not enforced at all, but are generally frowned upon to change for little to no reason:
> * Spaces after control statements
> 	* `if()` and `if ()` - nobody cares!

https://github.com/tgstation/tgstation/blob/master/.github/CONTRIBUTING.md#enforced-not-enforced

**Review by Twaticus — APPROVED** [review]

**Review by Maurukas — APPROVED** [review]
I can't comment on the code, but in terms of mapping, this will let us build airlocks with more flexible designs without rendering them completely ineffective.

**Comment by MrStonedOne** [comment]
@tgstation/commit-access

File: code/game/machinery/doors/airlock.dm
Comment (Cyberboss) [inline code/game/machinery/doors/airlock.dm:356]: .Cut() the list

File: code/game/machinery/doors/airlock.dm
Comment (Cyberboss) [inline code/game/machinery/doors/airlock.dm:388]: 2 ticks?

**Review by Cyberboss — CHANGES_REQUESTED** [review]

File: code/game/machinery/doors/airlock.dm
Comment (MrStonedOne) [reply code/game/machinery/doors/airlock.dm:356]: Reread these lines @Cyberboss

**Comment by LemonInTheDark** [comment]
> 
> 
> @tgstation/commit-access

Bruh

**Comment by MrStonedOne** [comment]
The stale system is a crap fix to a crap problem that I have never agreed with. 

When I notice things slip thru the cracks I'll speak up, because I still think the stale bot is bad.

File: code/game/machinery/doors/airlock.dm
Comment (MrStonedOne) [reply code/game/machinery/doors/airlock.dm:356]: wait, @Cyberboss are you suggesting they change the for to a `.Cut()` or add a Cut below it.

File: code/game/machinery/doors/airlock.dm
Comment (AlinaStarkova) [reply code/game/machinery/doors/airlock.dm:388]: im LITERALLY just doing the exact same method of closing that the current mapping helper method uses as coded by @MrStonedOne i have no idea what the timer delay is for, thats just how it currently works and i see no reason to change it

File: code/game/machinery/doors/airlock.dm
Comment (AlinaStarkova) [reply code/game/machinery/doors/airlock.dm:356]: no idea what that means, you would have to explain. im just doing the same thing the current helper does

**Comment by AlinaStarkova** [comment]
> 
> 
> The stale system is a crap fix to a crap problem that I have never agreed with.
> 
> When I notice things slip thru the cracks I'll speak up, because I still think the stale bot is bad.

ive been posting the link for this PR in coding channel 1-2 times a day lol, the only maintainer who ever looked at it was Timberpoes but he wanted a maptainer to approve it, and once a maptainer did timber just forgot.... i just want this merged before i get a mapmerge error or something, its such a simple helper that literally just builds upon code that already exists, if theres problem with the way my helper functions, then thats a problem with the original helper as well lol 😢

File: code/game/machinery/doors/airlock.dm
Comment (Cyberboss) [reply code/game/machinery/doors/airlock.dm:356]: close_others.Cut after for loop

Comment (AlinaStarkova) [reply code/game/machinery/doors/airlock.dm:356]: > 
> 
> close_others.Cut after for loop

otherlock.close_others.Cut(src)  ?

Comment (Cyberboss) [reply code/game/machinery/doors/airlock.dm:356]: No, src.close_others, which has references to other airlocks

File: code/game/machinery/doors/airlock.dm
Comment (MrStonedOne) [reply code/game/machinery/doors/airlock.dm:388]: mainly for that reason. It was designed to allow dragging something behind you without the door closing. also because its intentionally not as secure as the virology version.

File: code/game/machinery/doors/airlock.dm
Comment (MrStonedOne) [reply code/game/machinery/doors/airlock.dm:356]:
```suggestion
	if(close_others) //remove this airlock from the list of every linked airlock
		closeOtherId = null
		for(var/obj/machinery/door/airlock/otherlock as anything in close_others)
			otherlock.close_others -= src
		close_others.Cut()
	if(id_tag)
```

**Comment by optimumtact** [comment]
> The stale system is a crap fix to a crap problem that I have never agreed with.
> 
> When I notice things slip thru the cracks I'll speak up, because I still think the stale bot is bad.

We don't have the necessary staff to keep up with the tracker, and we can't scale up fast because the quality of talent is so shallow

File: code/game/machinery/doors/airlock.dm
Comment (AlinaStarkova) [reply code/game/machinery/doors/airlock.dm:356]: hmmm, im not sure exactly what this even does but ill add it in and test to ensure it doesnt cause any issues, doubt it will but testing is always good

**Comment by AlinaStarkova** [comment]
https://user-images.githubusercontent.com/40489693/129984711-dff12103-d12b-4795-bb12-786bee057a5e.mp4

**Pull request merged**
